### PR TITLE
DOCS-549: Updated info about keywords case-sensivity.

### PR DESCRIPTION
**Pull request by BayoNet**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Documentation

### CODE REVIEW

File: docs/en/query_language/syntax.md [hunk 1]
```diff
@@ -29,7 +29,14 @@ Comments in C-style: from `/*` to `*/`. These comments can be multiline. Spaces
 
 ## Keywords {#syntax-keywords}
 
-Keywords (such as `SELECT`) are not case-sensitive. Everything else (column names, functions, and so on), in contrast to standard SQL, is case-sensitive.
+Keywords are not case-sensitive when they correspond to:
```
Comment (blinkov) [inline docs/en/query_language/syntax.md]:
```suggestion
Keywords are case-insensitive when they correspond to:
```

File: docs/en/query_language/syntax.md [hunk 2]
```diff
@@ -29,7 +29,14 @@ Comments in C-style: from `/*` to `*/`. These comments can be multiline. Spaces
...
+
+- SQL standard. For example, `SELECT = select`, `AS = as`.
```
Comment (blinkov) [inline docs/en/query_language/syntax.md]:
```suggestion
- SQL standard. For example, `SELECT`,  `select` and `SeLeCt` are all valid.
```

File: docs/en/query_language/syntax.md [hunk 3]
```diff
@@ -29,7 +29,14 @@ Comments in C-style: from `/*` to `*/`. These comments can be multiline. Spaces
...
+- Implementation in some popular DBMS (MySQL or Postgres). For example, `DateTime = datetime`.
```
Comment (blinkov) [inline docs/en/query_language/syntax.md]:
```suggestion
- Implementation in some popular DBMS (MySQL or Postgres). For example, `DateTime` is same as  `datetime`.
```

File: docs/en/query_language/syntax.md [hunk 4]
```diff
@@ -29,7 +29,14 @@ Comments in C-style: from `/*` to `*/`. These comments can be multiline. Spaces
...
+
+Case-sensitivity of data type names you can check in the `system.data_type_families` table.
```
Comment (blinkov) [inline docs/en/query_language/syntax.md]:
```suggestion
Whether data type name is case-sensitive can be checked in the `system.data_type_families` table.
```

link to system table?

File: docs/en/query_language/syntax.md [hunk 5]
```diff
@@ -29,7 +29,14 @@ Comments in C-style: from `/*` to `*/`. These comments can be multiline. Spaces
...
+
+All other keywords (column names, functions, and so on), in contrast to standard SQL, are case-sensitive.
```
Comment (blinkov) [inline docs/en/query_language/syntax.md]:
```suggestion
In contrast to standard SQL all other keywords (including functions names)  are **case-sensitive**.
```
column name is an identifier, not a keyword

**Review by blinkov — CHANGES_REQUESTED** [review]

File: docs/en/query_language/syntax.md (hunk 4 above)
Comment (BayoNet) [reply docs/en/query_language/syntax.md]: It is not described. I'll do it later.

**Review by blinkov — APPROVED** [review]

**Pull request merged**